sql/core/src/test/scala/org/apache/spark/sql/sources/DDLTestSuite.scala

```diff
@@ -113,4 +113,23 @@ class DDLTestSuite extends DataSourceTest with SharedSQLContext {
     assert(attributes.map(_.name) === Seq("col_name", "data_type", "comment"))
     assert(attributes.map(_.dataType).toSet === Set(StringType))
   }
+
+  test("SPARK-7012 Create table statement should support NOT NULL modifier for columns") {
+    withTempPath { dir =>
+      val path = dir.getCanonicalPath
+      sql(
+      s"""
+         |CREATE TEMPORARY TABLE tempTableDDL
+         |( tCol1 INT NOT NULL,
+         |  tCol2 STRING
+         |)
+         |USING parquet
+         |OPTIONS (
+         |  path '$path'
+         |)
+       """.stripMargin
+      )
+      caseInsensitiveContext.dropTempTable("tempTableDDL")
```

Comment:
improve this test case to make sure we really make that column not nullable, i.e. read the table out and check its schema.